framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -682,6 +763,37 @@ public void setRoot(@Nullable CompilationUnitTree root) {
             // contents won't change between compilation units.
             // elementCache.clear();
         }
+
+        if (root != null && checker.hasOption("ajava")) {
+            // TODO: This relies on the toString implementations of MemberSelectTree and
+            // IdentifierTree. This could be turned into its own method that does this correctly.
```

Comment:
What is "this" in "does this correctly"?  Can you add a brief description of what this code does?